### PR TITLE
917/refactor data catalog

### LINKED ISSUE #917: Refactor data catalogs for v1
### Kind of request

Changing existing functionality

### Enhancement Description

Current catalogs are of course made to fit the v0.x API. These should be refactored to the new changes.

### Use case

Use the catalog with v1.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Jaapel**
## Issue addressed
Fixes #917 

## Explanation
Explain how you addressed the bug/feature request, what choices you made and why.

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation
- [x] Updated changelog.rst

## Data/Catalog checklist
- [x] `data/catalogs/predefined_catalogs.yml` has not been modified.
- [x] None of the old `data_catalog.yml` files have been chagned
- [ ] `data/chagnelog.rst` has been updated
- [ ] new file uses `LF` line endings (done automatically if you used `update_versions.py`)
- [x] New file has been tested locally
- [x] Tests have been added using the new file in the test suite

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

**Comment by Jaapel** [comment]
@savente93 There is still a test failing because I updated the data catalog. The registry file needs a hash. Is that a commit hash?

**Comment by savente93** [comment]
iirc it is the `sha256` of the catalog file, though note that old versions should not be updated for backwards comparability

**Review by savente93 — COMMENTED** [review]
One question but otherwise LGTM will approve when that is resolved.

File: hydromt/data_source/data_source.py [hunk 1]
```diff
@@ -73,12 +77,26 @@ def _validate_data_type(cls, data: Any) -> Any:
...
+    def full_uri(self) -> str:
+        """Join root with uri."""
+        uri_is_url: bool = is_valid_url(self.uri)
+        if uri_is_url:
+            # uri is fully self-describing
+            return self.uri
+        if not uri_is_url and self.root:
+            if is_valid_url(self.root):
+                # use '/' to connect url parts
+                return f"{self.root.rstrip('/')}/{self.uri}"
```
[the remark this answers is not in the document]
Comment (Jaapel) [reply hydromt/data_source/data_source.py:90]: What URIs that are not Windows Path will not use `/`? Windows paths are handled above.

[the remark this answers is not in the document]
Comment (savente93) [reply hydromt/data_source/data_source.py:90]: ah okay, sorry, if windows paths are handled earlier, I missed that. mea culpa. all good then

**Review by savente93 — APPROVED** [review]
LGTM

**Pull request merged**